backend/dal/Repositories/ActivityRepository.cs

```diff
@@ -0,0 +1,88 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Security.Claims;
+using MapsterMapper;
+using Microsoft.Data.SqlClient;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
+using Pims.Core.Extensions;
+using Pims.Dal.Entities;
+using Pims.Dal.Entities.Models;
+using Pims.Dal.Helpers.Extensions;
+
+namespace Pims.Dal.Repositories
+{
+    /// <summary>
+    /// Provides a repository to interact with research files within the datasource.
+    /// </summary>
+    public class ActivityRepository : BaseRepository<PimsActivityInstance>, IActivityRepository
+    {
+        #region Constructors
+        /// <summary>
+        /// Creates a new instance of a ResearchFileRepository, and initializes it with the specified arguments.
+        /// </summary>
+        /// <param name="dbContext"></param>
+        /// <param name="user"></param>
+        /// <param name="service"></param>
+        /// <param name="logger"></param>
+        public ActivityRepository(PimsContext dbContext, ClaimsPrincipal user, IPimsRepository service, ILogger<ResearchFileRepository> logger, IMapper mapper) : base(dbContext, user, service, logger, mapper) { }
+        #endregion
+
+        #region Methods
+
+        /// <summary>
+        /// Retrieves the activity  with the specified  id.
+        /// </summary>
+        /// <param name="id"></param>
+        /// <returns></returns>
+        public PimsActivityInstance GetById(long id)
+        {
+            return this.Context.PimsActivityInstances.AsNoTracking()
```

Comment:
lets move the AsNoTracking to the last statement before FirstOrDefault. That way none of the child entities are tracked either

Reply:
done